### PR TITLE
Survey Post handlers

### PR DESCRIPTION
**Pull request by MaianaB**
<p><b>Feature/Bug description:</b> Handlers for survey post. Methods get, post and delete</p>

<p><b>Solution:</b> Create files and tests</p>

<p><b>TODO/FIXME:</b> n/a</p>

### CODE REVIEW

File: backend/handlers/survey_post_collection_handler.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        data = json.loads(self.request.body)
+        institution_key = data['institution']
+        institution = ndb.Key(urlsafe=institution_key).get()
+
+        Utils._assert(institution.state == 'inactive',
+                      "The institution has been deleted",
+                      NotAuthorizedException)
+
+        Utils._assert(not user.has_permission("publish_survey_post", institution_key),
+                      "You don't have permission to publish post.",
```
Comment (mayzabeel) [inline backend/handlers/survey_post_collection_handler.py]: to publish survey

File: backend/handlers/survey_post_collection_handler.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+from utils import json_response
+
+from handlers.base_handler import BaseHandler
+from models.survey_post import SurveyPost
+from service_messages import send_message_notification
+
+from custom_exceptions.notAuthorizedException import NotAuthorizedException
+
+
+class SurveyPostCollectionHandler(BaseHandler):
```
Comment (mayzabeel) [inline backend/handlers/survey_post_collection_handler.py]: ver viabilidade de usar os handlers de post

File: backend/models/factory_post.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    def get_type(data):
+        """Return type of post according attributes that data has."""
+        if(data.get("type_survey")):
+            return 'SURVEY_POST'
+        else:
+            return 'POST'
+
+    @staticmethod
+    def get_permission(data):
+        """Return permission according type of data has."""
```
Comment (andreldsa) [inline backend/models/factory_post.py]: Essa sentença terminando com 'has' está estranho: http://sana.aalto.fi/awe/style/sentence/endverb/index.html

Sugestão: "Return permission according to the type of data"

File: backend/models/factory_post.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        if(PostFactory.get_type(data) == 'SURVEY_POST'):
+            return 'publish_survey_post'
```
Comment (andreldsa) [inline backend/models/factory_post.py]: publish_survey

File: frontend/notification/notificationDirective.js [hunk 5]
```diff
@@ -23,6 +23,10 @@
                 icon: "inbox",
                 state: "app.post"
             },
+            "SURVEY_POST": {
+                icon: "inbox",
```
Comment (andreldsa) [inline frontend/notification/notificationDirective.js]: icon: "poll"

File: backend/models/factory_post.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    @staticmethod
+    def get_type(data):
+        """Return type of post according attributes that data has."""
+        if(data.get("type_survey")):
+            return 'SURVEY_POST'
+        else:
+            return 'POST'
+
+    @staticmethod
+    def get_permission(data):
```
Comment (mayzabeel) [inline backend/models/factory_post.py]: mover esse método pro handler

File: backend/test/post_collection_handler_test.py [hunk 7]
```diff
@@ -202,3 +243,24 @@ def initModels(cls):
     cls.event.end_time = datetime.datetime.now()
     cls.event.local = "Event location"
     cls.event.put()
+
+    # Survey post
+    cls.options = [
+        {'id': 0,
+         'text': 'frist option',
```
Comment (andreldsa) [inline backend/test/post_collection_handler_test.py]: first*

**Pull request merged**